File: dagger2-scopes-and-subcomponents-part3/app/src/main/java/com/example/dagger2scopesandsubcomponentspart3/WarriorScreenComponent.java
package com.example.dagger2scopesandsubcomponentspart3;

import dagger.Component;
import dagger.Subcomponent;

@WarriorScreenScope
@Subcomponent(modules = {WarriorScreenModule.class})
public interface WarriorScreenComponent {

    void inject(WarriorActivity warriorActivity);
}

File: dagger2-scopes-and-subcomponents2/app/src/main/java/com/example/dagger2scopesandsubcomponentspart2/WarriorScreenComponent.kt
package com.example.dagger2scopesandsubcomponentspart2

import dagger.Component
import dagger.Subcomponent

//@WarriorScreenScope
//@Component(modules = [WarriorScreenModule::class],
//    dependencies = [AppComponent::class])
//interface WarriorScreenComponent {
//    fun inject(warriorActivity: WarriorActivity)
//}


@WarriorScreenScope
@Subcomponent(modules = [WarriorScreenModule::class])
interface WarriorScreenComponent {
    fun inject(warriorActivity: WarriorActivity)
}